File: z-clients/javascript/src/models/svixSinkConfig.ts
// this file is @generated
/**
* Configuration for a Svix sink. Each message is forwarded as a Svix message-create call
* (`POST {server_url}/api/v1/app/{app_id}/msg/`). This is a thin convenience over an HTTP sink.
* The `app_id`, `event_type`, and `payload` are templates rendered per-message
* (see [`diom_core::template_str`]).
*/
export interface SvixSinkConfig {
    /** Svix API token, sent as the bearer credential. Obfuscated in list responses. */
    token: string;
    /** Target Svix application. Can be optionally templated. */
    appId: string;
    /** Svix event type. Can be optionally templated. */
    eventType: string;
    /** Templated message payload. When absent, the raw message value bytes are used (must be JSON). */
    payload?: string | null;
    /**
     * Templated Svix `Idempotency-Key`. When absent or it renders to an empty string, a stable
     * key derived from the sink and message identity (namespace, topic, consumer_group, partition,
     * offset) is used so retries are de-duplicated by Svix.
     */
    idempotencyKey?: string | null;
    /** Optional base URL override. When absent, the region is inferred from the token. */
    serverUrl?: string | null;
}

export const SvixSinkConfigSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): SvixSinkConfig {
        return {
            token: object['token'],
            appId: object['app_id'],
            eventType: object['event_type'],
            payload: object['payload'],
            idempotencyKey: object['idempotency_key'],
            serverUrl: object['server_url'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: SvixSinkConfig): any {
        return {
            'token': self.token,
            'app_id': self.appId,
            'event_type': self.eventType,
            'payload': self.payload,
            'idempotency_key': self.idempotencyKey,
            'server_url': self.serverUrl,
        };
    }
}

File: z-clients/rust/src/models/svix_sink_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// Configuration for a Svix sink. Each message is forwarded as a Svix message-create call
/// (`POST {server_url}/api/v1/app/{app_id}/msg/`). This is a thin convenience over an HTTP sink.
/// The `app_id`, `event_type`, and `payload` are templates rendered per-message
/// (see [`diom_core::template_str`]).
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct SvixSinkConfig {
    /// Svix API token, sent as the bearer credential. Obfuscated in list responses.
    pub token: String,

    /// Target Svix application. Can be optionally templated.
    pub app_id: String,

    /// Svix event type. Can be optionally templated.
    pub event_type: String,

    /// Templated message payload. When absent, the raw message value bytes are used (must be JSON).
    #[serde(skip_serializing_if = "Option::is_none")]
    pub payload: Option<String>,

    /// Templated Svix `Idempotency-Key`. When absent or it renders to an empty string, a stable
    /// key derived from the sink and message identity (namespace, topic, consumer_group, partition,
    /// offset) is used so retries are de-duplicated by Svix.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub idempotency_key: Option<String>,

    /// Optional base URL override. When absent, the region is inferred from the token.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub server_url: Option<String>,
}

impl SvixSinkConfig {
    pub fn new(token: String, app_id: String, event_type: String) -> Self {
        Self {
            token,
            app_id,
            event_type,
            payload: None,
            idempotency_key: None,
            server_url: None,
        }
    }

    pub fn with_payload(mut self, value: impl Into<Option<String>>) -> Self {
        self.payload = value.into();
        self
    }

    pub fn with_idempotency_key(mut self, value: impl Into<Option<String>>) -> Self {
        self.idempotency_key = value.into();
        self
    }

    pub fn with_server_url(mut self, value: impl Into<Option<String>>) -> Self {
        self.server_url = value.into();
        self
    }
}
